### ISSUE TITLE
Allow maximizing/reverting the terminal pane

### ISSUE THREAD

**Issue by costincaraivan** [Microsoft/vscode#47661]
I'd be surprised if this hasn't been asked before, but tmux supports a thing it calls "pane zooming", if I recall correctly. 

Basically once you start splitting a window, vertically, horizontally, it doesn't matter, you press a certain keyboard shortcut and the current pane is "maximized", hiding all the other panes. Text is reflowed, so you actually see the former pane as a full window. This is incredibly useful once you launch some command that has a ton of unexpected output that you want to check out.

When you're done you press the same keyboard shortcut and the pane "reverts", showing the previous layout. The text in the pane is reflowed again to make it fit in the smaller pane size.

Very, very handy when you're moving through terminals quickly and you want to check out something.

**Comment by Tyriar** [comment]
This has not been asked for before but I'm aware of the feature. You currently have the ability to maximize the *panel* (`workbench.action.toggleMaximizedPanel`), not a split within the terminal.

**Comment by costincaraivan** [comment]
I know about that toggle, but it's not the same thing 😄 

Offtopic: And that toggle doesn't really do what it says on the tin 😛

**Comment by Tyriar** [comment]
@FancyBanana I think the OP wants the active terminal split pane to go full screen, including the side bar. Your issue is a problem though which I'll fix up now.

**Comment by Tyriar** [comment]
@FancyBanana coming in 1.28 https://github.com/Microsoft/vscode/commit/aacaecc68ac0c53dd595ba20dfc969a186a3139c

**Comment by costincaraivan** [comment]
Yes, I want the active terminal split pane to go full screen, filling at least the terminal side bar to 100%.

Ideally, if the terminal evolves and can be placed anywhere, including a normal tab, the terminal will have the ability to be maximized and terminal splits will also have this ability (+ the ability to be restored to their previous size).

Check out tmux for clarification: https://sanctum.geek.nz/arabesque/zooming-tmux-panes/

**Comment by glowtape** [comment]
Maximized terminal pane would be nice. VS Code makes as a nice host for a Powershell console.

**Comment by robclancy** [comment]
I use vscode in i3 with 3 terminals tiled below. I'm moving to inbuilt terminal now that it has splitting but I used to fullscreen a terminal often when doing things like `git status` or `git diff`. So that's my use case for wanting proper terminal fullscreen.

**Comment by Tyriar** [comment]
FYI @sbatten, this could be generalized for editors and the other panels as well.

**Comment by umdstu** [comment]
I too was looking for this functionality as well (or rather, in a roundabout way). I had been attempting to move a split terminal back to its own panel (uniquely accessible via the active terminals dropdown that exists) so that I could then see more of the context (and then could optionally use the existing terminal "Maximize Panel Size" functionality to see even more of it) .

Please consider adding this!

Thanks

**Comment by zljubisic** [comment]
I would like to have a key binding to maximize terminal to full screen with no VS code interface. Something like I have just started a new "Windows terminal". With the same key binding terminal should be restored to its previous size.

**Comment by erikbjoern** [comment]
In my experience there is more similar keyboard shortcuts missing, not least such a small thing as resetting the side panel width (just like it behaves on mouse double click).

Having the ability to in-/decrement the size like we can with the terminal would also be great. Especially if there were, as many have requested before, the option to decide for yourself how large the steps should be.

This could also be implemented as "resize currently focused pane" and the ability to move focus between different sections could be improved.

**Comment by Gerschel** [comment]
I'm a heavy vim/tmux user. I've got my vim keybindings setup in vscode, and tmux in my terminal. I'm very accustomed to ctrl+b z to zoom into a pane, and out. Sometimes in vim I go zen mode. So in a terminal, I can already zoom in and zen 80% wide from 10% off the left, using two keybindings. I put in my vote for this to be added to vscode. Vscode has been the only contender that could pull me away from my setup (that I can scp into any environment or use remotely).

**Comment by shankara-n** [comment]
Is this planned to be done?

**Comment by carlosflorencio** [comment]
Very good demo of this functionality:

https://github.com/microsoft/vscode/assets/1500881/748d901b-c2ae-42ff-9088-a383f8119463

From https://github.com/anuvyklack/windows.nvim.

Instead of having this feature builtin in the editor, with the right exposed apis we can have an extension providing this feature.

Some APIs I think are missing and required for this:
- Control the editors size (width and height)
- More accurate event listeners when switching editors (e.g `window.onDidChangeActiveEditorGroup()`)
  - The existing `window.onDidChangeActiveTextEditor()` might be enough for a hacky solution but it's not ideal because when it fires the current state of the `window.tabGroups.activeTabGroup` is still outdated iirc)
- Command to open the sidebar (currently there is only `workbench.action.toggleSidebarVisibility` and `workbench.action.closeSidebar`)

**Comment by madprops** [comment]
Only having horizontal splits that can't be maximized makes the terminal too cramped and hard to work with.

**Comment by Chealer** [comment]
>[…]
> 
> Check out tmux for clarification: https://sanctum.geek.nz/arabesque/zooming-tmux-panes/ (there's even a nice animation showing exactly what I mean)

Thank you for this report and this clarification @costincaraivan 
Since it may take a few more years before this is implemented and this comment clarifies your request a lot, I recommend you integrate these details in your original report.

Since `workbench.action.toggleMaximizedPanel` already allows maximizing a (single) terminal as long as it is not split, this issue’s title is unclear. I suggest retitling to something like:

> Split terminals/editors: No toggle to temporarily "zoom in" (maximize) on single terminal or editor side

<details>
  <summary>License</summary>
This comment and any other I might add to this ticket is offered under the terms of CC0 1.0.
</details>

**Comment by costincaraivan** [comment]
@Chealer ok, done, but it's been 7 years. Unlikely this is ever going to be added, but you never know.

**Comment by Chealer** [comment]
Thanks @costincaraivan but I do not find the new title much clearer. It is already possible to make an editor pane (for example) fullscreen. Moreover, since this is an ITS, tickets should be issue reports rather than literal requests.

It is already possible to have a terminal as a normal tab. What do you mean by "the terminal side bar"?

**Comment by costincaraivan** [comment]
You know what, I don't care anymore. I created this issue 7 years ago. Create a new issue with what you need.
